### PR TITLE
[LLM] BigDL-LLM Documentation Initial Version

### PR DESCRIPTION
**Pull request by Oscilloscope98**
## Description

Add BigDL-LLM documentation. https://github.com/analytics-zoo/nano/issues/534

Preview:
- Main page: https://yuwentestdocs.readthedocs.io/en/bigdl-llm-documentation/
- LLM Index page: https://yuwentestdocs.readthedocs.io/en/bigdl-llm-documentation/doc/LLM/index.html

Currently, the basic structure of the document has been added, as well as initial contents mostly copied from what we already have in README, tutorials, etc.

### CODE REVIEW

**Comment by shane-huang** [comment]
1. remove CLI in cards and index
2. change `model support` to `examples`
3. add GPU support in Key features
4. separate CPU/CPU in `installation`
5. HuggingFace Transformers API

**Comment by Oscilloscope98** [comment]
> 1. remove CLI in cards and index
> 2. change `Model Support` to `Examples`
> 3. add GPU support in Key features
> 4. separate CPU/GPU in `installation`
> 5. hide `Tips and Known Issue`
> 6. HuggingFace Transformers API

Has updated based on the comments :)

For GPU/CPU installation, I have added some initial contents. The GPU-related contents may need some further verification :)

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+* Intel Data Center GPU Flex Series
+
+To apply Intel GPU acceleration, there're several steps for tools installation and environment preparation:
+
+* Step 1, only Linux system is supported now, Ubuntu 22.04 is prefered.
+* Step 2, please refer to our [drive installation](https://dgpu-docs.intel.com/driver/installation.html) for general purpose GPU capabilities.
+  ```eval_rst
+  .. note::
+
+    IPEX 2.0.110+xpu requires Intel® oneAPI Base Toolkit's version >= 2023.2.0.
```
Comment (rnwang04) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:31]: This note should be moved to step 3.

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+  ```
+* Step 3, you also need to download and install [Intel® oneAPI Base Toolkit](https://www.intel.com/content/www/us/en/developer/tools/oneapi/base-toolkit-download.html). OneMKL and DPC++ compiler are needed, others are optional.
+  ```eval_rst
+  .. note::
+
+    IPEX 2.0.110+xpu requires Intel GPU Driver version is `Stable 647.21 <https://dgpu-docs.intel.com/releases/stable_647_21_20230714.html>`_.
```
Comment (rnwang04) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:37]: This note should be moved to step2.

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+
+BigDL-LLM for GPU supports has been verified on:
+
+* Intel Arc™ A-Series Graphics
+* Intel Data Center GPU Flex Series
+
+To apply Intel GPU acceleration, there're several steps for tools installation and environment preparation:
+
+* Step 1, only Linux system is supported now, Ubuntu 22.04 is prefered.
+* Step 2, please refer to our [drive installation](https://dgpu-docs.intel.com/driver/installation.html) for general purpose GPU capabilities.
```
Comment (rnwang04) [inline docs/readthedocs/source/doc/LLM/Overview/install_gpu.md:27]: driver installation

File: docs/readthedocs/source/doc/LLM/Overview/install_gpu.md (hunk 1 above)
Comment (Oscilloscope98) [reply docs/readthedocs/source/doc/LLM/Overview/install_gpu.md]: Thank you :) have updated based on comments

**Comment by jason-dai** [comment]
Please update the gif links and merge the first version

**Review by plusbang — APPROVED** [review]
LGTM

**Comment by Oscilloscope98** [comment]
@jason-dai We merged this PR for now as initial version for the document, with updated gif urls hosted on readthedocs :)

**Pull request merged**